src/include/duckdb/execution/operator/csv_scanner/global_csv_state.hpp

```diff
@@ -75,6 +75,8 @@ struct CSVGlobalState : public GlobalTableFunctionState {
 
 	atomic<idx_t> last_file_idx;
 	shared_ptr<CSVBufferUsage> current_buffer_in_use;
+
+	unordered_map<idx_t, idx_t> threads_per_file;
```

Comment:
Can we add a test with `SET temp_directory=''` which should prevent the creation of any temporary files and run the gzipped test from the issue?